lib/directory.coffee

```diff
@@ -94,6 +94,16 @@ class Directory
       for ignoredPattern in @ignoredPatterns
         return true if ignoredPattern.match(filePath)
 
+    if atom.config.get('tree-view.hidePattern')
+      basePath = path.basename(filePath)
+      hidePattern = atom.config.get('tree-view.hidePattern')
+      try
+        match = basePath.match(hidePattern)
+      catch
+        console.warn("Bad hidePattern regex: " + hidePattern)
+        return false
+      return true if match
+
```

Comment:
CoffeeScript supports string interpolation, so this might be more clear:

```coffee
console.warn("Bad hidePattern regex: #{hidePattern}")
```

You don't need to have the match variable, it could be rewritten this way:

```coffee
try
  return basePath.match(hidePattern)
catch
  console.warn("Bad hidePattern regex: #{hidePattern}")
  return false
```

Also, rather than using `console.warn`, you should probably use the new [notifications package](https://github.com/atom/notifications).

I'd also probably rewrite the section to not call `atom.config.get` twice:

```coffee
hidePattern = atom.config.get('tree-view.hidePattern')
return unless hidePattern
```

`atom.config.get` isn't a performance hog, but still.